approval_polls/views.py

```diff
@@ -412,21 +412,21 @@ def post(self, request, *args, **kwargs):
                     {'question_error': 'The question is missing'}
                 )
 
-            c = 1
-            name = 'choice1'
-            linkname = 'linkurl-choice1'
-
-            while (name in request.POST):
-                text = request.POST[name].strip()
-                if (text):
+            for key in request.POST:
+                # this could be done using String.startswith instead of re
+                # but then it would be harder to avoid POST params
+                # that aren't choices but happen to start with choice.
+                # in case someone adds a "choiceType" option later.
+                m = re.match("choice(\d+)", key)
+                if m:
+                    text = request.POST[key].strip()
                     choices.append(text)
```

Comment:
The app allows blank options to be added as valid choices. Perhaps you should check if the `text` choice is empty before appending it to the `choices` (what line 421 did).